modules/network-observability-virtualization-configuration.adoc

```diff
@@ -0,0 +1,120 @@
+// You can identify network flows coming from virtual machines connected to additional network interfaces that are using the Network Observability CNI plugin.
+
+ 
+.Prerequisites
+* Access to an {product-title} cluster with an additional network interface, such as a secondary interface or an L2 network.
+* Any other prerequisites?
+
+.Procedure
+. Get information about the virtual machine by running the following command. This information is used in Step 2: 
++
+[source,terminal]
+----
+oc get network-attachment-definitions.k8s.cni.cncf.io/netdevice67619-1 -o yaml
+----
+// NOTE to tech review: 
+// does any of this need to be anonymized for the customer? Or made to a <user_replaceable_value>? and is it necessary to use a different oc get command for different interfaces? Like I know the one above is from Mehul's SR-IOV example. But then in the QE test, its `$ oc get pod/virt-launcher-test-vm1-bsfb4 -o jsonpath='{.metadata.annotations.k8s\.v1\.cni\.cncf\.io/network-status}' | jq`. I can split this up and put both, but just not sure and want to verify. 
+. Investigate each additional network interface to find relevant pod annotations. These annotations differ depending on the kind of network interface you have: 
+.. For SR-IOV, look for the annotated fields:
++
+[source,yaml]
+----
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: openshift.io/netpolicy67619
+  creationTimestamp: "2024-10-03T20:40:48Z"
+  generation: 1
+  name: netdevice67619-1
+  namespace: openshift-sriov-network-operator
+  resourceVersion: "159492"
+  uid: 13131f06-e7b1-4b42-8d95-ac7c53e908cb
+spec:
+  config: |-
+    {
+        "cniVersion": "1.0.0",
+        "name": "netdevice67619-1", <1>
+        "type": "sriov",
+        "vlan": 0,
+        "vlanQoS": 0,
+        "logLevel": "info",
+        "ipam": {
+            "type": "static",
+            "addresses": [
+                {
+                    "address": "192.168.122.71/24" <2>
+                }
+            ]
+        }
+----
+<1> Name of the virtual virtual machine connected to the additional network interface.
+<2> The MAC address to identify the virtual machine.
+
+.. For L2, look for the annotated fields:
++
+[source,yaml]
+----
+ # ...
+  {
+    "name": "ovn-kubernetes",
+    "interface": "eth0",
+    "ips": [
+      "10.129.2.39"
+    ],
+    "mac": "0a:58:0a:81:02:27",
+    "default": true,
+    "dns": {}
+  },
+  {
+    "name": "my-vms/l2-network",   <1>
+    "interface": "podc0f69e19ba2",
+    "mac": "02:fb:f8:00:00:12",    <2>
+    "dns": {}
+  }
+ # ... 
+----
+<1> The name of the virtual machine connected to the additional network interface.
+<2> The MAC address to identify the virtual machine.
+
+. Configure `FlowCollector` based on the information you found from the additional network investigation. 
```

Comment:
This whole section is not needed to me.

You can directly configure the secondary network discovery without checking at these.

Reply:
```suggestion
. Get information about the virtual machine launcher pod by running the following command. This information is used in Step 2: 
+
[source,terminal]
----
$ oc get pod virt-launcher-<vm-name>-<random-suffix> -n <namespace> -o yaml
----
+
[source,yaml]
----
$ oc get pod virt-launcher-fedora-aqua-fowl-13-zr2x9 -n my-vms -o yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    k8s.v1.cni.cncf.io/network-status: |- 
      [{
        "name": "ovn-kubernetes",
        "interface": "eth0",
        "ips": [
          "10.129.2.39"
        ],
        "mac": "0a:58:0a:81:02:27",
        "default": true,
        "dns": {}
      },
      {
        "name": "my-vms/l2-network",   <1>
        "interface": "podc0f69e19ba2", <2>
        "mac": "02:fb:f8:00:00:12",    <3>
        "dns": {}
      }]
  name: virt-launcher-fedora-aqua-fowl-13-zr2x9
  namespace: default
spec:
  ...
status:
  ...
----
<1> The name of the secondary network.
<2> The network interface name of the secondary network..
<3> The MAC address used for secondary network.

. Configure `FlowCollector` based on the information you found from the additional network investigation. 
```

Reply:
> You can directly configure the secondary network discovery without checking at these.

just configuring name of secondary network wouldn't be enough, right? I think some sort of indexing would definitely be needed for enrichment, no?

Reply:
Yes @memodi, you need to specify the fields that define the index. However, you don't really care about the values here.

Either we let the user check the `k8s.v1.cni.cncf.io/network-status` and figure out by themself which fields are reliable or we directly tell them for each type of secondary network which fields to use as index.